packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -548,14 +616,39 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
 
     if (boolArg('pub')) {
       await pub.get(context: PubContext.create, directory: directory.path, offline: boolArg('offline'));
-      await project.ensureReadyForPlatformSpecificTooling(checkProjects: false);
+      await project.ensureReadyForPlatformSpecificTooling(checkProjects: pluginExampleApp);
+    }
+    if (templateContext['android'] == true) {
+      gradle.updateLocalProperties(project: project, requireAndroidSdk: false);
     }
-
-    gradle.updateLocalProperties(project: project, requireAndroidSdk: false);
-
     return generatedCount;
   }
 
+  void _updateTemplateContextWithPlatforms(Map<String, dynamic> context, List<String> platforms) {
+    for (final String platform in platforms) {
+      switch (platform) {
+        case 'ios':
+          context['ios'] = true;
+          break;
+        case 'android':
+          context['android'] = true;
+          break;
+        case 'web':
+          context['web'] =  featureFlags.isWebEnabled && true;
```

Comment:
Rather than conditionally checking here, you _might_ be able to only add web, linux, macos, windows to the list itself if the experiment is enabled. Then you can do:

```
for (final String platform in platforms) {
  context[platform] = true;
}
```

Reply:
Updated with a slightly different approach, please take a look and see if you like it.